molvis/molvisbase/include/inviwo/molvisbase/datastructures/molecularstructure.h

```diff
@@ -151,6 +173,29 @@ namespace molvis {
  */
 class IVW_MODULE_MOLVISBASE_API MolecularStructure {
 public:
+    /**
+     * \brief data structure holding the atom indices of a single backbone segment including
+     * dihedral angles
+     */
+    struct BackboneSegment {
+        bool valid() const {
+            return ca.has_value() && n.has_value() && c.has_value() && o.has_value();
```

Comment:
Why are they optional if they have to be there to be valid? Are they only optional during construction then?

Reply:
rename to `complete()`, add also add `empty()`